src/producer/mod.rs

```diff
@@ -129,6 +134,25 @@ impl<S: Read + Write> Producer<S> {
             .issue(&QueueControl::new(QueueAction::Resume, queues))?
             .await_ok()
     }
+
+    /// Initiate a new batch of jobs.
+    #[cfg(feature = "ent")]
+    pub fn start_batch(&mut self, batch: Batch) -> Result<BatchHandle<'_, S>, Error> {
+        let bid = self.c.issue(&batch)?.read_bid()?;
+        Ok(BatchHandle::new(bid, self))
+    }
+
+    /// Open an already existing batch of jobs.
+    #[cfg(feature = "ent")]
+    pub fn open_batch(&mut self, bid: String) -> Result<BatchHandle<'_, S>, Error> {
+        let bid = self.c.issue(&OpenBatch::from(bid))?.read_bid()?;
```

Comment:
Should we error if the returned `bid` is not `==` the provided `bid`?

Reply:
If the e2e tests, we are expecting the provided `bid` to be equal to the one returned. But now I am not sure why would do so in the first place. I guess it was just out of common sense (or out of desire to assert something :)), BUT this is not something they guarantee in the [docs](https://github.com/contribsys/faktory/wiki/Ent-Batches#batch-open-bid). At the same time, in the example in the docs they are using the _provided_ bid (rather than a returned one), when adding extra jobs to the re-opened batch. 

In the go [client](https://github.com/contribsys/faktory/blob/main/client/batch.go#L187-L203),
they are using the returned one and are not running the `==` check.

So the options are:
1) Throw away the returned bid and use the provided one for the extra jobs;
2) Follow the go bindings approach;

But I will definitely need to remove those assertions from the e2e tests.

Reply:
I'm generally inclined to follow the go bindings where there is ambiguity, since that is (as far as I can tell) the reference implementation maintained by the Faktory folks.